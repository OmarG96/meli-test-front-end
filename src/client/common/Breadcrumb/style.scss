@import "../../utils/scss/variables.scss";

.breadcrumb-root {
  grid-column: 2 / span 10;
  max-width: 1024px;
  width: 100%;
  margin: 16px auto 0;
}

.breadcrumb-route {
  font-size: 14px;
  color: $gray-6;

  &:not(:first-child) {
    &::before {
      content: "\003E";
      margin: 0 4px;
    }
  }
}
