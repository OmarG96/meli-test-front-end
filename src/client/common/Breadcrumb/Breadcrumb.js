import React from "react";
import { arrayOf, string } from "prop-types";

import "./style.scss";

const Breadcrumb = ({ path = [] }) => {
  return (
    <div className="breadcrumb-root">
      {path.map((route) => (
        <span className="breadcrumb-route">{route}</span>
      ))}
    </div>
  );
};

Breadcrumb.propTypes = {
  path: arrayOf(string),
};

export default Breadcrumb;
