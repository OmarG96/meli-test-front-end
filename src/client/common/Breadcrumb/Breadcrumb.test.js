import React from "react";
import Breadcrumb from ".";
import { render } from "@testing-library/react";

describe("GIVEN Breadcrumb", () => {
  describe("WHEN is rendered", () => {
    it("THEN match to snapshot", () => {
      const path = ["example1", "example2", "example3"];
      const { asFragment } = render(<Breadcrumb path={path} />);
      expect(asFragment()).toMatchSnapshot();
    });

    it("THEN match to snapshot when path is []", () => {
      const { asFragment } = render(<Breadcrumb />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
