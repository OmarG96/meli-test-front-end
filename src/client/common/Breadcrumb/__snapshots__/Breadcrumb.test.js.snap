// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN Breadcrumb WHEN is rendered THEN match to snapshot 1`] = `
<DocumentFragment>
  <div
    class="breadcrumb-root"
  >
    <span
      class="breadcrumb-route"
    >
      example1
    </span>
    <span
      class="breadcrumb-route"
    >
      example2
    </span>
    <span
      class="breadcrumb-route"
    >
      example3
    </span>
  </div>
</DocumentFragment>
`;

exports[`GIVEN Breadcrumb WHEN is rendered THEN match to snapshot when path is [] 1`] = `
<DocumentFragment>
  <div
    class="breadcrumb-root"
  />
</DocumentFragment>
`;
